oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/LogEntry2MetricsAdapter.java

```diff
@@ -81,10 +81,16 @@ public HTTPServiceMeshMetric.Builder adaptToUpstreamMetrics() {
         final long outboundStartTime = startTime + formatAsLong(properties.getTimeToFirstUpstreamTxByte());
         final long outboundEndTime = startTime + formatAsLong(properties.getTimeToLastUpstreamRxByte());
 
+        final HTTPServiceMeshMetric.Builder builder = adaptCommonPart();
+        // For client side call, status needs to be overridden because 4xx http status codes
+        // are considered as errors to.
+        final boolean status = builder.getResponseCode() < 400;
+
         return adaptCommonPart()
             .setStartTime(outboundStartTime)
             .setEndTime(outboundEndTime)
             .setLatency((int) Math.max(1L, outboundEndTime - outboundStartTime))
+            .setStatus(status)
```

Comment:
The status was not set? If so, the current status should be all `false`.
I am feeling there is a later place overriding this?

Reply:
> The status was not set? If so, the current status should be all `false`.
> I am feeling there is a later place overriding this?

It was actually set in `adaptCommonPart`, and here is to "override"

Reply:
OK, got it. Then this should be fine.